cellfinder_core/detect/detect.py

```diff
@@ -143,9 +158,18 @@ def main(
         processes.append(p)
         p.start()
 
-    processes[-1].join()
-    mp_3d_filter_queue.put((None, None, None))  # Signal the end
+    # Trigger callback when 3D filtering is done on a plane
+    nplanes_done = 0
+    while nplanes_done < len(signal_array):
+        callback(planes_done_queue.get(block=True))
+        nplanes_done += 1
+
+    # Wait for all the 2D filters to process
+    [p.join() for p in processes]
+    # Tell 3D filter that there are no more planes left
+    mp_3d_filter_queue.put((None, None, None))
```

Comment:
this looks a bit cryptic, could `(None, None, None)` be made a variable with a meaningful name?

Reply:
👍 but out of scope for this PR since this pre-existed? Perhaps worth opening an issue.

Reply:
Okay didn't realise it did: https://github.com/brainglobe/cellfinder-core/issues/38